### ISSUE TITLE
Shoutout not working at all

### ISSUE THREAD

**Issue by OGNetheri** [teklynk/twitch_shoutout#8]
I've tried to get this command to work, I've changed the settings on the web dashboard, changed the command name with or without exclamation mark, tried with personal and bot token. It just doesn't activate with any chosen command name. Of course i updated the source address, checked that OBS web source hw acceleration is on etc. 

So is this web source still running or am I still doing something wrong?

**Comment by teklynk** [comment]
The web source is still active and used by many. 
I recently tested this with !so and !clips as the commands. Verify that you Bot account is also a mod on your main channel.

Please try:

Test with very minimal settings: 
"so" as command.
Check: Show Clip.
Number of clips to pull = 20
Check: Show: "Go check out {channel}" on top of the clip
Max Duration Time = 10 secs
Check: Mods Only.

**Comment by OGNetheri** [comment]
So does this web source need users own bot to work? I thought as it's web source, it would work as Guru's web source works?

**Comment by teklynk** [comment]
Hmm. Are typing !so @user

**Comment by OGNetheri** [comment]
> Hmm. Are typing !so @user

"!so" is reserved to Guru's websource at this moment as it works. However earlier when I tried to get this to work I set e.g "!shoutout"(disabled the Guru's source for that) "!huuto" and couple other commands too (through web dashboard). No go. If it helps anyway, I'm using OBS v28.

**Comment by teklynk** [comment]
This is very strange. "!shoutout user" and "!shoutout @user" should work. 
I am assuming that the Twitch Account is set to your channel?
The only other suggestion would be to open the generated url in a browser and press F12 to bring up the dev tools. There you should see a Console tab where you can view the debug output and errors. I am curious if it is able to connect to your Twitch channel/chat and if it is able to pull a clip from https://twitchapi.teklynk.com/getuserclips.php?channel=YourChannelName&limit=100

**Comment by OGNetheri** [comment]
Yep, account is set for my channel 😁. 
Here's what I get with the basic settings with the !shoutout set as the command and with a minor "Custom Title".

![image](https://user-images.githubusercontent.com/110769602/235327737-0d801d75-b6c7-4329-8b47-ca9df4b9cb45.png)

**Comment by teklynk** [comment]
That looks good. Thanks for testing that. 

With the console window open, try to do a shoutout to see if any errors occur.

**Comment by OGNetheri** [comment]
![image](https://user-images.githubusercontent.com/110769602/235328413-d600343a-3875-4194-85e3-efbc7ea1952a.png)


And no go on video. Before you ask, yes, I've set the layout same as Guru is setup in OBS, so it should be seen. Also the custom CSS is set as suggested on https://twitch-shoutout.pages.dev/ settings (  OBS browser source Custom CSS section). Also tested without it, no change.

**Comment by OGNetheri** [comment]
One thing though. There's the "https://twitch-shoutout.pages.dev/shoutout.html? " start in the generated link as the server. As you had "[https://twitchapi.teklynk.com/" on you earlier post. So, should the server be the latter or is the generated links address correct?

**Comment by teklynk** [comment]
Thanks again.
Under the Network tab, do you see any errors?

https://twitch-shoutout.pages.dev/shoutout.html is correct. When you do a shoutout, the script will make a call out to https://twitchapi.teklynk.com to grab a clip. I am curious if maybe twitchapi.teklynk.com is being blocked by your ISP, firewall, pi-hole, something...

If anything is being blocked, it should appear in the network tab when you do a shoutout.

This appears to be a unique issue as no one else has reported it and I am not able to replicate the issue on my end. 

Here is what a successful shoutout should look like in the console log. 
![image](https://user-images.githubusercontent.com/4500737/235328698-b192e73c-395c-4360-8f09-7afe9d07ec36.png)

**Comment by OGNetheri** [comment]
![image](https://user-images.githubusercontent.com/110769602/235328836-ea46dc24-3bfb-40a1-8ce5-5b6f11dd787b.png)

**Comment by teklynk** [comment]
Thank you. 
Now try to do a shoutout with the network tab open to see if anything comes back as red. 

btw, the beacon.min.js is fine if it is blocked. It is not needed for the shoutouts.

**Comment by OGNetheri** [comment]
> Thank you. Now try to do a shoutout with the network tab open to see if anything comes back as red.
> 
> btw, the beacon.min.js is fine if it is blocked. It is not needed for the shoutouts.

That screenshot was taken with trying to !shoutout my own channel.

**Comment by OGNetheri** [comment]
![image](https://user-images.githubusercontent.com/110769602/235329044-f3ecb754-48b2-4440-8f29-392e84a32c87.png)

That's all i got, same as earlier.

**Comment by teklynk** [comment]
Hmm, strange. It's like it doesn't recognize your custom command. 

In the Custom Command field, type: shoutout
Then execute the command in twitch chat with: !shoutout @username
Regenerate the browser source url. 

Example:
![image](https://user-images.githubusercontent.com/4500737/235329241-252a219b-6b6a-446a-9645-ce8028f2aa48.png)

**Comment by OGNetheri** [comment]
That did the trick. So basically in the dashboard it has to be without exclamation mark and in twitch you must still use the exclamation mark. As I tried and it only now worked with "!" but not without. Excellent anyway, now I get to set up the automation! 

p.s. It might be a good to add on the dashboard instructions, that "!" is needed before the selected command, even when it's not written on the dashboard.

**Comment by teklynk** [comment]
Great. Glad that worked. Thanks for your time troubleshooting that. 

Yep. I have been told that I should update the instructions to be more clear. 

I will add something about using "!" to execute the command in chat or provide an example.

**Comment by OGNetheri** [comment]
Nice!

And thank you for  helping this out! If I can give yet another suggestion, it would be possibility to adjust the clips title pane, And if it would be adjusted only by code, then some example beside the pane's custom code, how to set tesxts location to e.g. low left or top left etc. to us who doesn't code anything. 😁

**Comment by teklynk** [comment]
Position options is a good idea. I will look into that this weekend. It may just be something simple like: top-left, top-right, bottom-left, bottom-right, center, bottom-center, top-center type of options. I could also add that for the title/streamer name text as well.

**Comment by OGNetheri** [comment]
> Position options is a good idea. I will look into that this weekend. It may just be something simple like: top-left, top-right, bottom-left, bottom-right, center, bottom-center, top-center type of options. I could also add that for the title/streamer name text as well.

Exactly what I was thinking. Thanks for the great websource(s)!!
